docs/design/core/metrics/prototype/MetricEvents.java

```diff
@@ -25,25 +25,49 @@
 import software.amazon.awssdk.metrics.meter.Timer;
 
 /**
- * Object used to store metric events collected by the SDK.
+ * An immutable object used to store metric events collected by the SDK.
  */
 @SdkPublicApi
 public interface MetricEvents {
     /**
-     * Add the given metric with associated data.
-     * @throws IllegalStateException If the given event is already present.
-     * @throws IllegalArgumentException If {@code eventData} is {@code null}.
+     * Return the metric data associated with the given event. Returns {@code
+     * null} if no event is found.
      */
-    <T> void putMetricEvent(MetricEvent<T> event, T eventData);
+    <T> T getMetricEventData(MetricEvent<T> event);
+
 
     /**
-     * Return the metric data associated with the given event. Returns {@code
-     * null} if no event is found.
+     * Return an iterator of the contained metric events and their data.
      */
-    <T> T getMetric(MetricEvent<T> event);
+    Iterator<MetricEventRecord<?>> getMetricEventsIterator();
+
+
+   /**
+    * A container associating an event with its data.
+    */
+   interface MetricEventRecord<T> {
```

Comment:
Should we move this to the top level of the package?

Reply:
Sure